binding.cc

```diff
@@ -390,8 +390,7 @@ namespace zmq {
 
   void
   Socket::UV_PollCallback(uv_poll_t* handle, int status, int events) {
-    assert(status == 0);
-
+    if (status != 0) NanThrowRangeError("I/O status: socket not ready !=0 ");
```

Comment:
(sorry i've had a few drinks) 

i'll make the changes :) I'm :+1: on the double negation. i remember when i wrote that I wanted to ensure the exact logic even at the expense of style. 

this area of the repo is very sensitive and will break the builds/tests if not done right